src/gui/addnewtorrentdialog.cpp

```diff
@@ -675,6 +675,7 @@ void AddNewTorrentDialog::setupTreeview()
         if (m_torrentInfo.filesCount() > 1) {
             m_contentModel = new TorrentContentFilterModel(this);
             connect(m_contentModel->model(), SIGNAL(filteredFilesChanged()), SLOT(updateDiskSpaceLabel()));
+            connect(m_contentModel->model(), SIGNAL(filteredFilesChanged()), SLOT(prioritizeFiles()));
```

Comment:
Not rational to do so.
We just need to do it once before we call addTorrent(). Moreover, in this place we will be able to "manually" create all ".unwanted" folders and assign the required attributes.
Another option: we can assign the attributes of the folders after we add torrent to session.